spec/features/state_file/review_page_spec.rb

```diff
@@ -0,0 +1,151 @@
+require "rails_helper"
+
+RSpec.feature "Completing a state file intake", active_job: true do
+  include MockTwilio
+  include StateFileIntakeHelper
+
+  before do
+    allow_any_instance_of(Routes::StateFileDomain).to receive(:matches?).and_return(true)
+  end
+
+  StateFile::StateInformationService.active_state_codes.each do |state_code|
+    context "#{state_code.upcase}" do
+      it "allows user to navigate to income review page, edit an income form, and then navigate back to final review page", required_schema: "az" do
+        set_up_intake_and_associated_records(state_code)
+
+        intake = StateFile::StateInformationService.intake_class(state_code).last
+
+        visit "/questions/#{state_code}-review"
+
+        # Final review page
+        expect(page).to have_text I18n.t("state_file.questions.shared.review_header.title")
+        within "#income-info" do
+          click_on I18n.t("general.edit")
+        end
+
+        # Income review page
+        expect(page).to have_text I18n.t("state_file.questions.income_review.edit.title")
+        within "#w2s" do
+          click_on I18n.t("state_file.questions.income_review.edit.review_and_edit_state_info")
+        end
+
+        # W2 edit page
+        expect(page).to have_text strip_html_tags(I18n.t("state_file.questions.w2.edit.instructions_1_html", employer: intake.state_file_w2s.first.employer_name))
+        fill_in strip_html_tags(I18n.t("state_file.questions.w2.edit.box15_html")), with: "987654321"
```

Comment:
we could but i don't think of this test as being the place to cover that. if they w2 page doesn't work that should be caught in the form spec. i could actually just delete this line because the point of this test is to see that clicking continue takes you to the right place?